src/Cache.hs

```diff
@@ -0,0 +1,77 @@
+module Cache (createCache, cacheRead, cacheWrite, cacheInvalidate) where
+
+import Data.Map (Map)
+import qualified Data.Map as Map
+import Definitions
+
+data Cache = Cache {
+    cacheSize :: Int, 
+    associativity :: Int, 
+    blockSize :: Int, 
+    cacheContent :: Map (Int, Int) CacheBlock
+} deriving (Show)
+
+data CacheBlock = CacheBlock {
+    isBlockValid :: Bool, 
+    isBlockDirty :: Bool, 
+    blockContent :: [Int]
+} deriving (Show)
+
+type MemoryAddress = Int
+type IsReadHit = Bool
+type IsWriteHit = Bool
+
+main = print (
+    cacheInvalidate (
+        fst (
+            cacheRead (createCache 64 2 4) 0x00000001)
+        ) 
+    0x00000001)
+
+{-  Creates an empty cache with the specified cache size, associativity, and block size.
```

Comment:
Documentation comments in Haskell usually follow the form:

`-- |<Start of documentation>` so that auto-doc tools can auto format. I think I have some in `SimulatorCore.hs`

So this line would be:
```haskell
-- |Creates an empty cache with the specified cache size, associativity, and block size.
--  Returns the empty cache on successful execution
`